dsc/tests/dsc_functions.tests.ps1

```diff
@@ -1060,4 +1060,58 @@ Describe 'tests for function expressions' {
     $out = $config_yaml | dsc config get -f - | ConvertFrom-Json
     $out.results[0].result.actualState.output | Should -BeExactly $expected
   }
+
+  It 'json() works: <testInput>' -TestCases @(
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @{ name = 'John'; age = 30 })').name"; expected = 'John' }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @{ name = 'John'; age = 30 })').age"; expected = 30 }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @(1,2,3))')[0]"; expected = 1 }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @(1,2,3))')[2]"; expected = 3 }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject 'hello')')"; expected = 'hello' }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject 42)')"; expected = 42 }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject $true)')"; expected = $true }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject $false)')"; expected = $false }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject $null)')"; expected = $null }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @{ users = @( @{ name = 'Alice' }, @{ name = 'Bob' } ) })').users[0].name"; expected = 'Alice' }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @{ users = @( @{ name = 'Alice' }, @{ name = 'Bob' } ) })').users[1].name"; expected = 'Bob' }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @{ key = 'value' })').key"; expected = 'value' }
+    @{ testInput = "('$(ConvertTo-Json -Compress -InputObject @{ nested = @{ value = 123 } })').nested.value"; expected = 123 }
+  ) {
+    param($testInput, $expected)
+    $expression = "[json$($testInput -replace "'", "''")]"
+
+    $config_yaml = @"
+            `$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
+            resources:
+            - name: Echo
+              type: Microsoft.DSC.Debug/Echo
+              properties:
+                output: '$expression'
+"@
+    $out = $config_yaml | dsc config get -f - | ConvertFrom-Json
+    $out.results[0].result.actualState.output | Should -Be $expected
+  }
+
+  It 'json() error handling: <expression>' -TestCases @(
+    @{ expression = "[json('not valid json')]"; expectedError = 'Invalid JSON string' }
+    @{ expression = "[json('{""key"":""value""')]"; expectedError = 'Invalid JSON string' }
+    @{ expression = "[json('')]"; expectedError = 'Invalid JSON string' }
+    @{ expression = "[json('{incomplete')]"; expectedError = 'Invalid JSON string' }
+    @{ expression = "[json('[1,2,')]"; expectedError = 'Invalid JSON string' }
```

Comment:
If they all have the same error, you don't need to have `expectedError` in the test case